src/NUnitEngine/nunit.engine.core/RuntimeFramework.cs

```diff
@@ -641,6 +684,73 @@ private static void AddMonoFramework(Version frameworkVersion, string profile)
 
             _availableFrameworks.Add(framework);
         }
+
+        private static void FindDotNetCoreFrameworks()
+        {
+            const string WINDOWS_INSTALL_DIR = "C:\\Program Files\\dotnet\\";
+            const string LINUX_INSTALL_DIR = "/usr/shared/dotnet/";
+            string INSTALL_DIR = Path.DirectorySeparatorChar == '\\'
+                ? WINDOWS_INSTALL_DIR
+                : LINUX_INSTALL_DIR;
+
+            if (!Directory.Exists(INSTALL_DIR))
+                return;
+            if (!File.Exists(Path.Combine(INSTALL_DIR, "dotnet.exe")))
+                return;
+
+            string runtimeDir = Path.Combine(INSTALL_DIR, "shared\\Microsoft.NETCore.App");
+            if (!Directory.Exists(runtimeDir))
+                return;
+
+            var dirList = new DirectoryInfo(runtimeDir).GetDirectories();
+            var dirNames = new List<string>();
+            foreach (var dir in dirList)
+                dirNames.Add(dir.Name);
+            var runtimes = GetNetCoreRuntimesFromDirectoryNames(dirNames);
+
+            _availableFrameworks.AddRange(runtimes);
+        }
+
+        // Internal for testing
+        internal static IList<RuntimeFramework> GetNetCoreRuntimesFromDirectoryNames(IEnumerable<string> dirNames)
```

Comment:
This method looks exciting! 😂 Would you mind clarifying what's going on here? All the directory names in my install are simple version numbers, but the logic here makes me think there are other options?

Reply:
Comment at https://github.com/nunit/nunit-console/pull/937#discussion_r619679210 refers - resolving this one.